source/solvers/gls_navier_stokes.cc

```diff
@@ -1269,6 +1278,33 @@ GLSNavierStokesSolver<dim>::set_initial_condition_fd(
           this->pcout << "*********************************" << std::endl;
         }
     }
+  else if (initial_condition_type ==
+           Parameters::InitialConditionType::average_velocity_profile)
+    {
+      this->pcout << "*********************************************************"
+                  << std::endl;
+      this->pcout
+        << " Initial condition using average velocity from checkpoint "
+        << std::endl;
+      this->pcout << "*********************************************************"
+                  << std::endl;
+
+      // The average velocity profile needs to come from the results of a
+      // simulation. The checkpoint/restart mechanism is used to obtain all the
+      // simulations information necessary to restart the simulation. The same
+      // mesh needs to be used, but new physics can be added to the simulation.
+      this->read_checkpoint();
+
+      // This is an initial condition, not a restart. Therefore, the current
+      // time is set to 0, the iteration number as well and the pvd is cleared
+      // from previous results
+      this->simulation_control->set_current_time(0.0);
+      this->simulation_control->set_iteration_number(0);
+      this->simulation_control->set_current_time_step(
+        this->simulation_parameters.simulation_control.dt);
+
+      this->pvdhandler.times_and_names.clear();
```

Comment:
I am not sure, but maybe a better approach would be to have a simulation_control->reset_time() function which  would reset the time, iteration and dt to parameter it has read instead.
This would prevent this manual modification of the simulation control object from outside of it and make it clear inside of the simulation_control object what is being modified. I think this would be a better approach.